core/block/editor/dashboard.go

```diff
@@ -75,6 +78,40 @@ func (p *Dashboard) StateMigrations() migration.Migrations {
 	}})
 }
 
+// autoInstallBinWidget installs bin widget for the existing users if it was not added/removed before and the user has archived objects
+func (p *Dashboard) autoInstallFavoriteWidget() error {
```

Comment:
Looks like these two auto-install functions can reuse the same code and be called with a widgetObjectId argument

Reply:
actually it was wrong because we don't need to create it with the same logic as archive. We only need to create explicitly when calling objectSetIsFavorite